cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -389,6 +389,29 @@ private class ExplicitSingleFieldStoreQualifierNode extends PartialDefinitionNod
   }
 }
 
+private FieldAddressInstruction getFieldInstruction(Instruction instr) {
+  result = instr or
+  result = instr.(CopyValueInstruction).getUnary()
+}
+
+private class BufferMayWriteSideEffectFieldStoreQualifierNode extends PartialDefinitionNode {
```

Comment:
You don't get to add a class named `BufferMayWriteSideEffectFieldStoreQualifierNode` without giving it some very good QLDoc :wink:.

Reply:
Good point! I've added a QLDoc in c560c7584c4889bedc032fe53695e75b696f02ec. I don't know if the quality of the QLDoc justifies this horrible name, though.